mlir/lib/Dialect/Tensor/IR/TensorOps.cpp

```diff
@@ -2711,15 +2711,38 @@ struct InsertSliceOpCastFolder final : public OpRewritePattern<InsertOpTy> {
     auto dstType = llvm::dyn_cast<RankedTensorType>(dst.getType());
     if (!srcType || !dstType)
       return failure();
+
+    // The tensor.cast source could have additional static information not seen
+    // in the insert slice op static sizes, so we ignore dynamic dims when
+    // computing the rank reduction mask.
+    SmallVector<int64_t> staticSizes(insertSliceOp.getStaticSizes());
```

Comment:
The values won't be modified, how about using `ArrayRef<int64_t>` here?

Reply:
They actually do get modified later on L2735.

Reply:
I see, somehow I missed it when I was searching the use of the variable.